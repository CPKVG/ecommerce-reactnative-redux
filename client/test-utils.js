// import { configureStore } from '@reduxjs/toolkit';
// import React from 'react';
// import { render as rtlRender } from '@testing-library/react-native'
// import { Provider } from 'react-redux';
// import { PersistGate } from 'redux-persist/integration/react';
// import { persistor } from './redux/createStore';
// import { rootReducer } from './redux/rootReducer';

// //custom render

// function render(
//     ui ,{
//         preloadedState,
//         store = configureStore({ reducer: rootReducer, preloadedState}),
//         ...renderOptions
//     } = {}
// ){
//   function Wrapper( {childern} ){
//       return <Provider store = {store}>
//           <PersistGate persistor={persistor}>
//             {childern}
//           </PersistGate>
//       </Provider>
//   }  
//   return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
// }

// //re-export everything
// export * from '@testing-library/react-native'
// //override render method
// export { render }


import React from "react"
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
// import configureStore from 'redux-mock-store';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { rootReducer } from "./redux/rootReducer";

const INITIAL_STATE = { ...rootReducer }; // Or use value from root reducer
const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
// const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
  render(ui, {
    
    wrapper: ({ children }) => <
      Provider store={store}>
          {children}
      </Provider>,
    ...options
  });

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender as render, store };