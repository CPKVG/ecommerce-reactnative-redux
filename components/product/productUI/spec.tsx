import React from "react"

import { NavigationContainer } from '@react-navigation/native';
//import { render, waitFor,cleanup } from '@testing-library/react-native';
import { render } from "../../../test-utils";
import renderer from "react-test-renderer"

import { cleanup, fireEvent } from "@testing-library/react-native";
import { Text, TouchableOpacity, View } from "react-native";
import { navigationRef } from "../../../RootNavigation";
import ProductUI from ".";
// import { shallow, mount } from 'enzyme';


afterEach(cleanup)
jest.useFakeTimers()

describe('testing product UI component',() => {

    function mockInput(objData:any,result:Boolean){
        // if (navigationRef.isReady()) {
            // const sentHandler = jest.fn()
            // const { getByPlaceholderText } = render(<NavigationContainer><ProductUI/></NavigationContainer>)
            // render(<NavigationContainer><CheckoutUI validCheckoutInput = {sentHandler}/></NavigationContainer>)
         
            // Object.keys(objData).forEach((x:any) => {
            //     return fireEvent.changeText(getByPlaceholderText(x),objData[x]);
            // });
            // expect(sentHandler).toBe(result)
        // }
    }

    // test("it should return true if input is true",()=>{

    //     mockInput(inputPlaceholder,true)
    // }

    test("btn event changes screen to purchase" ,()=>{
        const component = (
            <NavigationContainer>
                <ProductUI/>
            </NavigationContainer>
        )
                const { getByTestId,findByText } = render(component)
                const button = getByTestId('navi_btn')
                
                fireEvent.press(button)
                const newScreen = findByText('Shopping Cart')
                expect(button).toBeTruthy
                expect(newScreen).toBeTruthy                
    })
    it("renders the Product UI component",() => {
        let count = 0
        const tree = renderer.create(
            <View>
            <TouchableOpacity>
                <Text>
                Add to Cart ({count})
                </Text>
            </TouchableOpacity>
            
            <TouchableOpacity>
                <Text>
                    To CheckOut
                </Text>
            </TouchableOpacity>
        </View>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })


})