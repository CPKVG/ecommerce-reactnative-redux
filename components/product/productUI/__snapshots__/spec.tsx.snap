// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing product UI component renders the Product UI component 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Add to Cart (
      0
      )
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      To CheckOut
    </Text>
  </View>
</View>
`;
